src/metabase/driver/mysql.clj

```diff
@@ -271,6 +272,13 @@
    ;; strip off " UNSIGNED" from end if present
    (keyword (str/replace (name database-type) #"\sUNSIGNED$" ""))))
 
+(defmethod sql-jdbc.sync/column->special-type :mysql
+  [_ database-type _]
+  ;; Used when syncing. If it's :JSON type then it's :type/SerializedJSON special-type
```

Comment:
Based on the documentation it sounds like a `JSON` column isn't really a `TEXT` column: https://dev.mysql.com/doc/refman/5.7/en/json.html – it sounds like it's stored as a binary representation. So I'm not sure these types make sense – wouldn't it make more sense to match the way we treat JSON columns in Postgres? 

Either way, we'll need some tests before we can merge this. Something like this at the very least https://github.com/metabase/metabase/blob/master/test/metabase/driver/postgres_test.clj#L66

Altho ideally I think we'd want another test that looks at the values in the results as well